autopush/websocket.py

```diff
@@ -540,12 +542,14 @@ def process_notifications(self):
 
         # Are we paused? Try again later.
         if self.paused:
-            self.deferToLater(1, self.process_notifications)
+            d = self.deferToLater(1, self.process_notifications)
+            d.addErrback(self.trap_cancel)
```

Comment:
Just to clarify: the deferred will still be rejected with a `CancelledError`, even if you pass a custom `canceller`?